games-simulation/openrct2/openrct2-0.2.4.ebuild

```diff
@@ -0,0 +1,182 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake-utils xdg-utils
+
+MY_PN="OpenRCT2"
+MY_PN_OBJ="objects"
+MY_PN_RPL="replays"
+MY_PN_TS="title-sequences"
+MY_PV_OBJ="1.0.12"
+MY_PV_RPL="0.0.4"
+MY_PV_TS="0.1.2c"
+
+DESCRIPTION="An open source re-implementation of Chris Sawyer's RollerCoaster Tycoon 2"
+HOMEPAGE="https://openrct2.org/"
+SRC_URI="
+	https://github.com/${MY_PN}/${MY_PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz
+	https://github.com/${MY_PN}/${MY_PN_OBJ}/releases/download/v${MY_PV_OBJ}/${MY_PN_OBJ}.zip -> ${PN}-${MY_PN_OBJ}-${MY_PV_OBJ}.zip
+	https://github.com/${MY_PN}/${MY_PN_TS}/releases/download/v${MY_PV_TS}/${MY_PN_TS}.zip -> ${PN}-${MY_PN_TS}-${MY_PV_TS}.zip
+	test? ( https://github.com/${MY_PN}/${MY_PN_RPL}/releases/download/v${MY_PV_RPL}/${MY_PN_RPL}.zip -> ${PN}-${MY_PN_RPL}-${MY_PV_RPL}.zip )
+"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="dedicated libressl +lightfx +multiplayer +opengl test +truetype twitch"
+REQUIRED_USE="dedicated? ( !lightfx multiplayer !opengl !truetype !twitch )"
+
+DEPEND_RDEPEND="
+	dev-libs/icu:=
+	dev-libs/jansson:=
+	dev-libs/libzip:0=
+	media-libs/libpng:0=
+	sys-libs/zlib:=
+	!dedicated? (
+		media-libs/libsdl2:=
+		media-libs/speexdsp:=
+		opengl? ( virtual/opengl )
+		truetype? (
+			media-libs/fontconfig:=
+			media-libs/freetype:2=
+			media-libs/sdl2-ttf:=
```

Comment:
I couldn't find a reference for this package. Where did you infer this dependency from? I see no `#include` for `SDL_ttf.h` anywhere and the package builds fine without it.

Reply:
Oh, good to know. Is `SDL_ttf.h` that one, which indicated the package being used? I wasn't here sure, so I did ask upstream, and they said, it's needed. Will drop it then.

Reply:
As also here request on IRC, let me try to clarify. My first lookout is be the build system, in case of cmake here these are `CMakeLists.txt` files, which I look into, which checks are implemented to find needed dependencies. The `sdl2-ttf` wasn't there listed. I guess, my false assumption is the description of the package `library that allows you to use TrueType fonts in SDL applications`, which lead, that it's needed for sdl2-truetype stuff. But that's wrong. Indeed, I also try to look into the source code, if there is something to be included, which needs to be present. I even grepped for *ttf* stuff, but didn't found anything. As of the description, I wasn't 100% sure, so I included it (since it's a small package), if maybe there is a case, I don't see and it could fail because of that.

Reply:
I think this is an upstream bug. I digged through the code and found this (https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/openrct2/drawing/TTFSDLPort.cpp#L2):

```
/**
 * The following code is from SDL2_ttf (2 Jan 2017).
 * Taking just what was needed for OpenRCT2 with all SDL2 calls
 * removed.
 */
```

Looks to me like they've adopted the code from libsdl2-ttf instead of linking against it. I created an issue on their github: https://github.com/OpenRCT2/OpenRCT2/issues/10308

Reply:
Funny, that they gived you a different answer than me :/ Allright, that seems to be clarified, as it can be dropped entirely.